src/geom/intersection_tools.C

```diff
@@ -150,6 +150,32 @@ bool within_edge(const Elem & elem,
   return false;
 }
 
+unsigned short at_vertex(const Elem & elem,
+                         const Point & p,
+                         const Real tol)
+{
+  for (const auto v : elem.vertex_index_range())
+    if (elem.point(v).relative_fuzzy_equals(p, tol))
```

Comment:
This isn't a proper relative tolerance.  Here `relative_fuzzy_equals` will use `p+elem.point(v)` to compare against, which is likely to be way too big in some cases (a tiny element far from the origin) and is going to be effectively *zero* in other cases (a huge element with a vertex at the origin).

I'd normally use `elem.hmax()` as a scale here, but that might be too expensive for you?  If so, it wouldn't be unreasonable to have something like `Elem::cheap_scale()` that returns something like an l1 norm (how do we *not* already have that for `TypeVector`...) of `(elem.point(1)-elem.point(0))`.  Kind of arbitrary for anisotropic meshes but so is the choice of `hmax()` over `hmin()`.